src/main/java/com/opentable/jvm/Nmt.java

```diff
@@ -0,0 +1,257 @@
+        return new Formatter().toString();
+    }
+
+    /**
+     * Requires JVM argument -XX:NativeMemoryTracking=summary.
+     * Logs a warning if there was an error getting the NMT summary or if NMT was disabled.
+     * @return null if there was an error getting the summary.
+     */
+    @Nullable
+    static Nmt get() {
+        final String nmt = invoke();
+        if (nmt == null) {
+            return null;
+        }
+        try {
+            return parse(nmt);
+        } catch (IllegalArgumentException e) {
+            LOG.warn("un-parseable NMT data:\n{}", nmt, e);
+            return null;
+        }
+    }
+
+    /**
+     * @param s String to parse.
+     * @return Filled-out {@link Usage} instance.
+     * @throws IllegalArgumentException with human-readable error if string couldn't be parsed.
+     */
+    @VisibleForTesting
+    static Usage parseUsage(final String s) {
+        final String reservedLabel = "reserved=";
+        final String firstKB = "KB, ";
+        int i, j;
+        i = s.indexOf(reservedLabel);
+        if (i == -1) {
+            throw new IllegalArgumentException("could not find reserved label");
+        }
+        j = s.indexOf(firstKB, i);
+        if (j == -1) {
+            throw new IllegalArgumentException("could not find KB after reserved label");
+        }
+        final String reservedStr = s.substring(i + reservedLabel.length(), j);
+        final long reserved;
+        try {
+            reserved = Long.parseLong(reservedStr) * K;
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(String.format("could not parse reserved %s", reservedStr));
+        }
+        final String committedLabel = "committed=";
+        i = s.indexOf(committedLabel, j + firstKB.length());
+        if (i == -1) {
+            throw new IllegalArgumentException("could not find committed label");
+        }
+        j = s.indexOf("KB", i);
+        final String committedStr = s.substring(i + committedLabel.length(), j);
+        final long committed;
+        try {
+            committed = Long.parseLong(committedStr) * K;
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(String.format("could not parse committed %s", committedStr));
+        }
+        return new Usage(reserved, committed);
+    }
+
+    /**
+     * @param nmt {@link #invoke()} Diagnostic command} output to parse.
+     * @return Filled-out {@link Nmt} instance.
+     * @throws IllegalArgumentException with human-readable error if string couldn't be parsed.
+     */
+    @VisibleForTesting
+    static Nmt parse(@Nonnull final String nmt) {
+        final String prefixTotal = "Total: ";
+        final String prefixDash = "-";
+        final String prefixParen = " (";
+
+        final List<String> lines = Arrays.asList(nmt.split("\n"));
+        if (lines.size() < 5) {
+            throw new IllegalArgumentException("insufficient lines to parse");
```

Comment:
might consider sending the actual data out as a warn or part of the exception message, it might be informative and help diagnose what went wrong